pyomo/repn/plugins/gams_writer.py

```diff
@@ -326,15 +333,17 @@ def _write_model(self,
                 if v.is_binary():
                     binary.append(var)
                 elif v.is_integer():
-                    if v.bounds == (0,1):
+                    if (v.has_lb() and (value(v.lb) >= 0)) and \
+                       (v.has_ub() and (value(v.ub) <= 1)):
                         binary.append(var)
                     else:
                         ints.append(var)
-                elif v.lb == 0:
+                elif value(v.lb) == 0:
                     positive.append(var)
                 else:
                     reals.append(var)
             else:
+                # GH: doesn't seem like this block is ever hit
```

Comment:
Ok, I see now. So this _used_ to be a necessary bit of code, back when the labeling was done through the `__str__` method (when I set the body variable to `str(con.body)` instead of using a StringIO with `con.body.to_string()`, which is what I do now). When that was the way I did it, expressions were replaced with a name in the constraint definitions in the GAMS file, and then those names were later defined to be the contents of the expression. This happened because Expressions inherit `__str__` from ComponentData just the same as variables, so expressions were labeled the same way as variables (by being replaced with a label), but then expressions were of course later defined fully. However, now labeling is done through `to_string`, and Expressions inherit `to_string` from `_ExpressionData`, which doesn't label it with a name but instead prints the expanded expression. So yeah, expressions are never added to `var_list` anymore.

I suppose then that means this block is no longer needed. However, I will say that this way of labeling and defining expressions is what @jsiirola implemented in his first iteration of the gams writer, but if we're ok with expressions being written out entirely in their constraints instead of existing as expressions that are referenced by name by their constraints, then I'm fine with that.